environment/types/types.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// Pre-defined environment keys. Environment settings can be added w/o being defined
+// in this section.
+const (
+	CategoriesKey = "categories"
+)
+
+// ErrNotFound is returned by Get() for missing keys
+var ErrNotFound = fmt.Errorf("environment key not found")
+
+// ApiCredentials is set of identifiers and secrets used to authenticate with
+// the underlying infrastructure.
+type ApiCredentials struct {
+	// Username for basic authentication
+	Username string
+	// Password for basic authentication
+	Password string
+	// KeyPair is JSON-encoded key pair for TLS client authentication
+	KeyPair string
+}
+
+// ManagementEndpoint specifies API endpoint used for interacting with underlying
+// infrastructure.
+type ManagementEndpoint struct {
+	// ApiCredentials embedded into endpoint
+	ApiCredentials
+	// Address is URL of managment endpoint
+	Address *url.URL
+	// Whether to authenticate TLS endpooint
+	Insecure bool
+}
+
+// Topology is a map of topological domains to topological segments.
+// A topological domain is a sub-division of a cluster, like "region",
+// "zone", "rack", etc
+type Topology map[string]string
```

Comment:
Should we define Topology value types as consts? 
```go
type TopologyItem string
type Topology map[TopologyItem]TopologyItem
const (
    PrismCentralZone TopolyItem = "..."
    PrismElementCluster = "..."
)
```
or do we expect topologies to be highly flexible based on the use-case?

Reply:
That's TBD. You're right we should standardize on topology keys. For instance an availability zone might correspond to one or more PC endpoints.

Reply:
Cool, we can tackle that in future PRs.